Refrigerator.cs

```diff
@@ -0,0 +1,238 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace ExerciseRefrigerator
+{
+    public class Refrigerator
+    {
+        public static int UniqIdFriger = 0;
+        public int Id { get; }
+        private string _model;
+        private string _color;
+        private int _numShelves;
+        public List<Shelf> Shelves { get; set; }
+        public Refrigerator()
+        {
+
+        }
+        public Refrigerator(string model, string color, int numShelves)
+        {
+            Id = UniqIdFriger++;
+            Model = model;
+            Color = color;
+            NumShelves = numShelves;
+            Shelves = new List<Shelf>();
+        }
+
+        public string Model
+        {
+            get
+            {
+                return this._model;
+            }
+            set
+            {
+                try
+                {
+                    if (value.Length <= 1) throw new Exception("invalide model");
+                    _model = value;
+                }
+                catch (Exception e)
+                {
+                    Console.WriteLine("error: " + e.Message);
+                }
+
+            }
+        }
+
+        public string Color
+        {
+            get
+            {
+                return this._color;
+            }
+            set
+            {
+                try
+                {
+                    if (value.Length <= 1) throw new Exception("invalide color");
+                    _color = value;
+                }
+                catch (Exception e)
+                {
+                    Console.WriteLine("error: " + e.Message);
+                }
+
+            }
+        }
+
+        public int NumShelves
+        {
+            get
+            { return _numShelves; }
+
+            set
+            {
+                try
+                {
+                    if (value <= 0) throw new Exception("invalide size");
+                    _numShelves = value;
+                }
+                catch (Exception e)
+                {
+                    Console.WriteLine("error: " + e.Message);
+                }
+            }
+        }
+
+        //2
+        public int GetFreeSpace()
+        {
+            int freeSpace = 0;
+
+            //for (int i = 0; i < NumShelves; i++)
+            //{
+            //    freeSpace += Shelves[i].GetCurrentFreeSpace();
+            //}
+            freeSpace = Shelves.Sum(item => item.GetCurrentFreeSpace() + freeSpace);
```

Comment:
This works, but only because freeSpace is 0. This isn't accumulate function, this is Sum + transform, where each item transforms to an integer and added.